client/src/modules/fee_center/modals/fee_center.modal.js

```diff
@@ -54,11 +55,24 @@ function FeeCenterModalController($state, FeeCenter, ModalService, Notify, AppCa
             vm.auxiliaryCenter = 1;
           }
           vm.feeCenter.is_cost = reference.is_cost;
-          vm.feeCenter.reference_cost_id = reference.account_reference_id;
-          vm.costCenterReference = {
-            account_reference_id : reference.account_reference_id,
-            is_cost : reference.is_cost,
-          };
+
+          if (reference.is_variable) {
+            vm.feeCenter.reference_cost_variable_id = reference.account_reference_id;
+            vm.variableCostCenterReference = {
```

Comment:
The `vm.variableCostCenterReference` and `vm.fixCostCenterReference` contain the same information.  When this happens, to save space, you should use an extra variable:
```js
const costCenterReferenceBundle = {
              account_reference_id : reference.account_reference_id,
              is_cost : reference.is_cost,
              is_variable : reference.is_variable,
              is_turnover : reference.is_turnover,
}

if (reference.is_variable) {
  vm.variableCostCenterReference  = costCenterReferenceBundle;
} else {
  vm.fixedCostCenterReference  = costCenterReferenceBundle;
}
```